sp/src/game/server/ai_senses.cpp

```diff
@@ -144,7 +144,11 @@ void CAI_Senses::Listen( void )
 
 bool CAI_Senses::ShouldSeeEntity( CBaseEntity *pSightEnt )
 {
+#ifdef EZ2
+	if ( pSightEnt == GetOuter() || (!(pSightEnt->GetFlags() & FL_OBJECT) && !pSightEnt->IsAlive()) )
```

Comment:
How can we be sure this won't have unintended side effects?

Reply:
`FL_OBJECT` is not assigned by and/or used by anything else except for missiles and, in E:Z2, item crates. Neither of them use `IsAlive()` in any capacity, except the latter overrides it to allow item crates to be seen by the sensing code. (which could theoretically be removed with this code in place)

Reply:
I haven't forgotten about this! I want to do a little bit of a deep dive into this tomorrow just for my own understanding. Once I've looked through the code and I feel like I understand it better, I'll approve this PR

Reply:
I've noticed that the drivable APC also is given the flag ``FL_OBJECT``. So, after this change, projectiles like flares, grenades, etc as well as drivable APCs and remarkables will pass "ShouldSeeEntity" when they would have failed before. Will this have any effect? I think that we should test with the APC after merging this PR. Just to make sure that NPCs don't act differently around projectiles, vehicles, etc. I don't know exactly how this function is used so I'm just trying to be cautious.